pkg/downloader/chart_downloader.go

```diff
@@ -97,7 +97,32 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 
 	name := filepath.Base(u.Path)
 	destfile := filepath.Join(dest, name)
-	if err := ioutil.WriteFile(destfile, data.Bytes(), 0644); err != nil {
+	tmpfile, err := ioutil.TempFile(dest, "tmp")
+	// Remove the temp file in case an error occurred before renaming.
+	defer func() {
+		if tmpfile != nil {
+			if _, err := os.Stat(tmpfile.Name()); err == nil {
+				os.Remove(tmpfile.Name())
```

Comment:
on Windows it won't remove the file unless we release the handler to the file. Can we call `tmpfile.Close()` here? There's a chance we can return on a bad write, causing the handle to never be released.